scene/main/canvas_item.cpp

```diff
@@ -557,37 +557,41 @@ void CanvasItem::draw_rect(const Rect2 &p_rect, const Color &p_color, bool p_fil
 		}
 
 		RenderingServer::get_singleton()->canvas_item_add_rect(canvas_item, p_rect, p_color);
+	} else if (p_width >= p_rect.size.width || p_width >= p_rect.size.height) {
+		RenderingServer::get_singleton()->canvas_item_add_rect(canvas_item, p_rect.grow(0.5f * p_width), p_color);
 	} else {
-		// Thick lines are offset depending on their width to avoid partial overlapping.
-		// Thin lines don't require an offset, so don't apply one in this case
-		real_t offset;
-		if (p_width >= 2) {
-			offset = p_width / 2.0;
-		} else {
-			offset = 0.0;
+		// Workaround for scaling issue. See `RendererCanvasCull::canvas_item_add_line()`.
+		if (p_width <= 1.001) {
+			p_width = 1.002;
 		}
 
+		real_t offset = 0.5f * p_width;
```

Comment:
```suggestion
		real_t offset = 0.5f * p_width;

		// Don't offset if line primitives instead of quads. See `RendererCanvasCull::canvas_item_add_line()`.
		if (p_width <= 1.001) {
			offset = 0.0;
		}
```

For reference: https://github.com/godotengine/godot/pull/69851#issuecomment-1345398511.

Reply:
The lines will overlap in the corners, which will be noticeable with a translucent color.

![](https://user-images.githubusercontent.com/47700418/206889271-7e93789e-5079-4945-8db1-fc666f3341b0.png)

The only option to remove this hack is #69851 or something similar. But I don't know how to make it work for thin lines (`width < 0`) in an easy way. In any case, `offset` must not be 0.